jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/dao/AbstractPipelineMavenPluginDao.java

```diff
@@ -1143,8 +1143,12 @@ protected String getDatabaseDescription() {
         }
     }
 
+    void updateBuildOnCompletion(@Nonnull String jobFullName, int buildNumber, int buildResultOrdinal, long startTimeInMillis, long durationInMillis) {
+        updateBuildOnCompletion(jobFullName, buildNumber, buildResultOrdinal,  Result.SUCCESS.ordinal == buildResultOrdinal, 0, 0);
```

Comment:
`Result.SUCCESS.ordinal == buildResultOrdinal`
this makes your test fail.

You have to make the same test as in `DatabaseSyncRunListener`